### PR TITLE
add redux and connect frontend to backend

### PR DESCRIPTION
**Pull request by hienlpv**

### CODE REVIEW

File: frontend/src/api/user.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import axios from 'axios';
+
+import { baseURI } from './constant';
+
+const fetchUsers = async () => {
+  try {
+    const user = await axios.get(baseURI + 'auth/get-all');
```
Comment (mahoangnhatphi) [inline frontend/src/api/user.js]: Please add paths to contants

File: frontend/src/components/Modal.jsx [hunk 2]
```diff
@@ -1,6 +1,22 @@
 import React, { useState, useRef } from 'react';
+import { useDispatch } from 'react-redux';
 import { Modal, Button, Container, Col, Row, Form } from 'react-bootstrap';
 
+import { createUser } from '../redux/action/user';
+
+const modalLabel = {
+  add: {
+    title: 'ADD NEW USER',
```
Comment (mahoangnhatphi) [inline frontend/src/components/Modal.jsx]: Move to constants

File: frontend/src/pages/dashboard/index.jsx [hunk 3]
```diff
@@ -46,55 +50,57 @@ function Dashboard() {
...
               <td>{index + 1}</td>
-              <td>{user.usrID}</td>
-              <td>{user.fullName}</td>
+              <td>{user.id}</td>
+              <td>{user.lastName + ' ' + user.firstName}</td>
               <td>{user.email}</td>
-              <td>{user.birthDate}</td>
-              <td>{user.firstLoginDate}</td>
-              <td>{user.lastLoginDate}</td>
+              <td>{new Date().toLocaleDateString()}</td>
```
Comment (mahoangnhatphi) [inline frontend/src/pages/dashboard/index.jsx]: Use date format lib to ready for adapting with any format

**Review by mahoangnhatphi — CHANGES_REQUESTED** [review]

File: frontend/src/components/Modal.jsx (hunk 2 above)
Comment (hienlpv) [reply frontend/src/components/Modal.jsx]: this is contants

File: frontend/src/components/Modal.jsx [hunk 4]
```diff
@@ -6,12 +6,12 @@ import { createUser } from '../redux/action/user';
 
 const modalLabel = {
   add: {
-    title: 'ADD NEW USER',
+    title: 'add new user',
     cancel: 'Cancel',
     submit: 'Save',
```
Comment (mahoangnhatphi) [inline frontend/src/components/Modal.jsx]: Move these labels to constants for reuse

**Review by mahoangnhatphi — CHANGES_REQUESTED** [review]

File: backend/src/logger/logs/errors.log [hunk 5]
```diff
@@ -722,3 +722,35 @@
 [2022-03-06 13:42:28] [[32minfo[39m]: ==================== AuthenticationRepository, call method GetAll ====================
```
Comment (mahoangnhatphi) [inline backend/src/logger/logs/errors.log]: Please add this file to `.gitignore`

**Review by mahoangnhatphi — CHANGES_REQUESTED** [review]

**Review by mahoangnhatphi — APPROVED** [review]
ok

**Pull request merged**